src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/AttributeSection.test.tsx

```diff
@@ -193,6 +219,77 @@ describe("AttributeSection", () => {
     const dateInput2 = await screen.findByTestId("CalendarIcon");
     expect(dateInput).toBeInTheDocument();
     expect(dateInput2).toBeInTheDocument();
+
+    fireEvent.change(dateInput, { target: { value: "01/01/2023" } });
+    expect(dateInput.value).toBe("01/01/2023");
+
+    const addButton = screen.getByTestId("AddCircleOutlineIcon");
+    expect(addButton).toBeInTheDocument();
+    userEvent.click(addButton);
+  });
+
+  it("shows RatioInput on selecting the Ratio type", async () => {
+    const assessmentElement: AssessmentPerformed = new AssessmentPerformed();
+    const { container } = render(
+      <AttributeSection selectedDataElement={assessmentElement} />
+    );
+
+    const attributeSelectBtn = screen.getByRole("button", {
+      name: "Attribute Select Attribute",
+    });
+    expect(attributeSelectBtn).toBeInTheDocument();
+
+    userEvent.click(attributeSelectBtn);
+
+    const attributeSelect = await screen.findByRole("listbox");
+    const attributeOptions = within(attributeSelect).getAllByRole("option");
+    expect(attributeOptions).toHaveLength(7);
+
+    userEvent.click(within(attributeSelect).getByText(/result/i));
+    const attributeInput = within(attributeSelectBtn.parentElement).getByRole(
+      "textbox",
+      { hidden: true }
+    );
+    expect(attributeInput).toBeInTheDocument();
+    expect(attributeInput).toHaveValue("Result");
+    const typeSelectBtn = await screen.findByRole("button", {
+      name: /type/i,
+    });
+    expect(typeSelectBtn).toBeInTheDocument();
+    userEvent.click(typeSelectBtn);
+    const typeSelect = await screen.findByRole("listbox");
+    expect(typeSelect).toBeInTheDocument();
+    const typeOptions = within(typeSelect).getAllByRole("option");
+    expect(typeOptions.length).toEqual(9);
+    fireEvent.click(within(typeSelect).getByText("Ratio"));
+
+    await waitFor(() => {
+      setTimeout(() => {
+        const quantityValueInput = screen.getByTestId(
+          "quantity-value-input-Ratio"
+        ) as HTMLInputElement;
+        expect(quantityValueInput).toBeInTheDocument();
+        fireEvent.change(quantityValueInput, { target: { value: "1" } });
+        expect(quantityValueInput.value).toBe("1");
+
+        const autocomplete = screen.getByTestId("quantity-unit-dropdown-Ratio");
+        const input = within(autocomplete).getByRole(
+          `combobox`
+        ) as HTMLInputElement;
+        autocomplete.click();
+        autocomplete.focus();
+        fireEvent.change(input, { target: { value: "wk" } });
+        act(async () => {
+          await new Promise((resolve) => setTimeout(resolve, 0));
+        });
+        fireEvent.click(screen.getAllByRole("option")[1]);
+        fireEvent.change(input, { target: { value: "/wk per week" } });
+
+        const addButton = screen.getByTestId("AddCircleOutlineIcon");
+        expect(addButton).toBeInTheDocument();
+        userEvent.click(addButton);
+      }, 6000);
+    });
```

Comment:
you can add a timeout for waitFor if needed

Reply:
I did try that, the test still failed